openid/src/main/java/fish/payara/security/openid/azure/AzureOpenIdExtension.java

```diff
@@ -0,0 +1,229 @@
+import fish.payara.security.annotations.OpenIdProviderMetadata;
+import fish.payara.security.openid.OpenIdExtension;
+import fish.payara.security.openid.OpenIdIdentityStore;
+import static fish.payara.security.openid.OpenIdUtil.getConfiguredValue;
+import static fish.payara.security.openid.OpenIdUtil.isEmpty;
+import fish.payara.security.openid.api.DisplayType;
+import fish.payara.security.openid.api.PromptType;
+import java.lang.annotation.Annotation;
+import java.util.ArrayList;
+import java.util.List;
+import static java.util.Objects.nonNull;
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.event.Observes;
+import javax.enterprise.inject.spi.AfterBeanDiscovery;
+import javax.enterprise.inject.spi.BeanManager;
+import javax.enterprise.inject.spi.BeforeBeanDiscovery;
+import javax.enterprise.inject.spi.CDI;
+import javax.enterprise.inject.spi.ProcessBean;
+import javax.security.enterprise.authentication.mechanism.http.HttpAuthenticationMechanism;
+import javax.security.enterprise.identitystore.IdentityStore;
+import org.eclipse.microprofile.config.Config;
+import org.eclipse.microprofile.config.ConfigProvider;
+
+/**
+ * Activates {@link AzureOpenIdAuthenticationMechanism} with the
+ * {@link AzureAuthenticationDefinition} annotation configuration.
+ *
+ * @author Gaurav Gupta
+ */
+public class AzureOpenIdExtension extends OpenIdExtension {
+
+    private final List<AzureAuthenticationDefinition> definitions = new ArrayList<>();
+
+    @Override
+    protected void beforeBeanDiscovery(@Observes BeforeBeanDiscovery beforeBeanDiscovery, BeanManager manager) {
+        addAnnotatedType(AzureOpenIdAuthenticationMechanism.class, manager, beforeBeanDiscovery);
+        super.beforeBeanDiscovery(beforeBeanDiscovery, manager);
+    }
+
+    /**
+     * Find the {@link AzureAuthenticationDefinition} annotation and validate.
+     *
+     * @param <T>
+     * @param bean
+     * @param beanManager
+     */
+    @Override
+    protected <T> void findOpenIdDefinitionAnnotation(@Observes ProcessBean<T> bean, BeanManager beanManager) {
+        AzureAuthenticationDefinition definition = bean.getAnnotated().getAnnotation(AzureAuthenticationDefinition.class);
+        if (nonNull(definition) && !definitions.contains(definition)) {
+            definitions.add(definition);
+            validateExtraParametersFormat(toOpenIdAuthDefinition(definition));
+        }
+    }
+
+    @Override
+    protected void afterBeanDiscovery(@Observes AfterBeanDiscovery afterBean, BeanManager beanManager) {
+        if (!definitions.isEmpty() && beanManager.getBeans(IdentityStore.class).isEmpty()) {
+            afterBean.addBean()
+                    .scope(ApplicationScoped.class)
+                    .beanClass(IdentityStore.class)
+                    .types(IdentityStore.class, Object.class)
+                    .createWith(obj -> CDI.current().select(OpenIdIdentityStore.class).get());
```

Comment:
`CDI.current()` doesn't play well with extensions, as it tries to find bean manager for calling class. Usually, the problem is when the call involves a class from user application. It is always more safe to look up in JNDI first. (I might file an issue, that our CDI provider should do that :)). And you already got applicable beanManager in your parameter.

Consider this applicable for any other use of `CDI.current()`.